Controllers/StudentsController.cs

```diff
@@ -28,5 +31,32 @@ public StudentsController(IStudentRepo studentService)
             // instead of 200(Ok)
             // return CreatedAtRoute(nameof(GetStudentById), new {Id = student.Id}, student);
         }
+
+        //GET api/students
+        [HttpGet]
+        public ActionResult <IEnumerable<Student>> GetStudents(){
+            var students = _studentService.GetAllStudent();
+            return Ok(students);
+        }
+
+        //GET api/students/{id}
+        [HttpGet("{id}", Name = "GetStudentById")]
+        public ActionResult <Student> GetStudentById(int id){
+            var student = _studentService.GetStudentById(id);
+            if(student != null){
+                return Ok(student);
+            }
+            return NotFound();
+        }
+
+        //GET api/students/{keyword}
+        [HttpGet("Search")]
+        public ActionResult <IEnumerable<Student>> SearchStudents([FromQuery] string keyword){
+            if(!string.IsNullOrWhiteSpace(keyword)){
+                var students = _studentService.FindStudents(keyword);
+                return Ok(students);
+            }
+            return BadRequest();
```

Comment:
what is the message of your bad request?

Reply:
I have added a descriptive error object to the bad request now. Thanks.